### DISCUSSION TITLE
Feature Request: Hide read articles

### DISCUSSION THREAD

**Discussion by OrkoGrayskull** [jocmp/capyreader#643] [Feature Requests]
Read articles are grayed out in the article overview. It would be nice if there was a filter option to hide the articles completely - so that only the unread ones appear.

When this could be useful: Every now and then RSS feeds are updated late and then appear further down with the articles that have already been read. With this option, you could make sure that no article is overlooked and only the unread ones are displayed.

**Comment by mzehrer** [comment]
I second this. I use capyreader on miniflux with the Google reader API. And sometimes I also use the mini flux web ui to read articles. It woul be nice to hide already read items.

**Comment by jocmp** [comment]
@OrkoGrayskull There's a dropdown in the drawer that will filter to unread articles only and apply that to all feeds and folders in the list, too. Unless you're thinking of something else?

<img src="https://github.com/user-attachments/assets/641fc9d7-e049-4548-808a-914764b8b268" width="300px">

**Comment by OrkoGrayskull** [comment]
OK. The option already exists - I just thought it was somewhere else. Thank you very much! 

Just one suggestion for improvement: When you tap the “Mark as read” checkbox, the view should be refreshed so that the unread articles are hidden immediately - regardless of a refresh.

**Reply by jocmp** [reply]
sure thing. and I added that "Mark as read" functionality to [the latest build](https://github.com/jocmp/capyreader/releases/tag/2024.12.1088-dev). It might be nice to open the drawer too when that happens or leave it on the empty feed. I'm open to suggestions!

**Reply by OrkoGrayskull** [reply]
Thank you. That works wonderfully.

**Reply by PhilC813** [reply]
I thought it was a bug introduced in v1088. Personally not a fan of that behaviour.

It's a bit inconsistent in terms of behaviour because if you read one article, it stays visible in the unread view, while mark all dismisses them. I like that read articles stay visible temporarily just in case I missed something I wanted to read.

It will be more relevant to have this feature if, as mentioned by @jocmp, the drawer opens simultaneously.

Maybe there could be an option to enable/disable this under Settings > General > Mark All As Read? It would make sense if this setting combined the dismissing of articles and the drawer opening.

**Reply by jocmp** [reply]
I think I'd want these to be exclusive options so that there's not a weird conflict where you can select "open next feed" and "open drawer" simultaneously. But the option list would then be

"After Read All"

- Do nothing
- Hide articles
- Open next feed

**Reply by PhilC813** [reply]
👆 Definitely the cleanest implementation

**Reply by jocmp** [reply]
I added this "After Read All" option in [2025.01.1089](https://github.com/jocmp/capyreader/releases/tag/2025.01.1089). The default is the pre-1088 behavior which does nothing. The "Open Navigation Drawer" option will clear articles and open the drawer. "Open Next Feed" is consistent with the previous behavior.

**Comment by OrkoGrayskull** [comment]
I have tested the latest version. There are currently these three options (different identifiers):
- Do nothing
- Hide articles & Open Navigation Drawer
- Hide articles & Open next feed

I would like to suggest a fourth option: 
- Hide articles & do nothing
